### PR TITLE
(fix/desktop) removing custom picture is not reflected on contact's side

### PR DESCRIPTION
**Pull request by kounkou**
### Description

This PR fixes [9947](https://github.com/status-im/status-desktop/issues/9947) and contains : 

- Commit to fix the changing of custom picture and having the change reflected on contact's side
- Commit to fix the deleting of picture and having the change reflected on contact's side

-------

### Design decisions :

Previously, the inconsistency in reflecting profile picture changes occurred due to a bug tied to the design choice for generating and serving images through URLs. Upon uploading an image, status-go generates an image and saves a URL link. The observed issue was that even after changing the profile image, the URL remained the same. Although the backend correctly associated the URL with the newly uploaded image, Qt/QML failed to update the value accordingly.

### Possible solutions :

- **[Preferred - Implemented]** Using the same link but with subtle parameter to help Qt/Qml update the underlying values
   - This approach adds a silent parameter (Timestamp)
   - It helps version the image in a way
   - It's not breaking the current design and is invisible to the view
  
- Generating different image URL links for each images
   - While this approach looks ideal, it's resource intensive
   - It represents a breaking change for the Status App
   - It requires some dive deep to understand the impact

-------

### Affected areas

- Chat, Image URL generation, Contact Profile updating

### Screenshot of functionality (including design for comparison)

[profile updates.webm](https://github.com/status-im/status-go/assets/2589171/60a24da8-3ff2-4efa-8d6f-b52f845c4e05)

### CODE REVIEW

File: src/app/modules/main/chat_section/chat_content/messages/module.nim [hunk 1]
```diff
@@ -563,11 +564,13 @@ method getNumberOfPinnedMessages*(self: Module): int =
 
 method updateContactDetails*(self: Module, contactId: string) =
   let updatedContact = self.controller.getContactDetails(contactId)
+
+  let updatedSenderIcon = if updatedContact.icon != "" : addTimestampToURL(updatedContact.icon) else: ""
```
Comment (MishkaRogachev) [inline src/app/modules/main/chat_section/chat_content/messages/module.nim:568]: suggestion: move `if icon != ""` conditions to `addTimestampToURL` utils

File: ui/imports/shared/views/ProfileDialogView.qml [hunk 2]
```diff
@@ -130,6 +130,11 @@ Pane {
         d.reload()
     }
 
+    function addTimestampToURL(url) {
```
Comment (MishkaRogachev) [inline ui/imports/shared/views/ProfileDialogView.qml:133]: There are 2 similar functions. Can you please move it to qml utils (`/ui/imports/utils/Utils.qml` should be fine)?

**Review by MishkaRogachev — COMMENTED** [review]
Nice PR, just one question

**Review by MishkaRogachev — APPROVED** [review]
Great!

File: ui/imports/utils/Utils.qml [hunk 3]
```diff
@@ -938,4 +938,12 @@ QtObject {
     function isPunct(c) {
         return /(!|\@|#|\$|%|\^|&|\*|\(|\)|\+|\||-|=|\\|{|}|[|]|"|;|'|<|>|\?|,|\.|\/)/.test(c)
     }
+
+    function addTimestampToURL(url) {
```
Comment (endulab) [inline ui/imports/utils/Utils.qml:942]: You have got the same function in two places: qml and nim.
Please, add a new function to global/utils.nim and use addTimestampToURL there. 
global/utils.nim object is exposed to qml. This way you will have this function in one place only.
Also, can you add a test/nim for this function?

**Review by endulab — CHANGES_REQUESTED** [review]

File: ui/imports/utils/Utils.qml (hunk 3 above)
Comment (kounkou) [reply ui/imports/utils/Utils.qml:942]: Fixed, I still trying to make work the adding of Unittest for the utils function as there was no infrastructure for the test. I need some more time to get the Makefile updated as I am getting link error while compile test file involving the Utils.nim file.

```
/usr/bin/ld: nimcache/debug/utils_test/@m..@s..@svendor@snim-status-go@sstatus_go.nim.c.o: in function `isAlias__OOZOOZvendorZnim45status45goZstatus95go_84':
/home/jacques/Documents/work/status-desktop/vendor/nim-status-go/status_go.nim:108: undefined reference to `IsAlias'
/usr/bin/ld: nimcache/debug/utils_test/@m..@s..@svendor@snim-status-go@sstatus_go.nim.c.o: in function 
...
```

In the meantime, the image URL received look like : 

```
https://Localhost:35651/contactImages?imageName=thumbnail&publicKey=0x04d76f84e8d041e733456e71eb811d486e16a51fd94fa8d5ffde9d39ed902317bc8ae2f632b562e716ce4b2fca1d7dc54420c412658167a7d0cecb2fb1de23407c
```

then we are just adding a `&timestamp=1231243242342` or `?timestamp=1231243242342` at the end.

Those query params are silent.

Comment (kounkou) [reply ui/imports/utils/Utils.qml:942]: nvm, thanks for your proposal offline to actually keep the time_utils.nim just call it from the global utils 👍🏾

**Review by endulab — APPROVED** [review]

**Pull request merged**